experimental/LieAlgebras/src/root_systems.jl

```diff
@@ -0,0 +1,224 @@
+mutable struct RootSystem
+  roots::Vector{Vector{Int}}
+  simple_roots::Vector{Vector{Int}}
+  positive_roots::Vector{Vector{Int}}
+  root_system_type::Tuple{Symbol,Int64}
+  GAP_root_system::GAP.GapObj
+  function RootSystem(S::Symbol, n::Int64)
+    # S is a symbol detailing the type of the indecomposable root system 
+    # e.g. "A", "B", "C",... and n is an integer for the number of simple roots
+    S1 = GAP.Obj(S)
+    RS = GAP.Globals.RootSystem(S1, n)
+    sR = Vector{Vector{Int}}(GAP.Globals.SimpleSystem(RS))
+    Ro1 = Vector{Vector{Int}}(GAP.Globals.PositiveRoots(RS))
+    Ro2 = Vector{Vector{Int}}(GAP.Globals.NegativeRoots(RS))
+    Ro = vcat(Ro1, Ro2)
+    t = (S, n)
+    return new(Ro, sR, Ro1, t, RS)
+  end
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+@doc raw"""
+    number_of_roots(R::RootSystem)
+
+Return the numbers of roots in the root system `R`.
+"""
+number_of_roots(R::RootSystem) = size(R.roots)[1]
+
+@doc raw"""
+    number_of_roots(S::Symbol, n::Int64)
+
+Return the numbers of roots in the root system of type `S`
+"""
+number_of_roots(S::Symbol, n::Int64) = number_of_roots(RootSystem(S, n))
+
+@doc raw"""
+    getindex(R::RootSystem, r::Int)
+
+Return the `r`-th root of the root system `R`.
+"""
+getindex(R::RootSystem, r::Int) = getindex(R.roots, r)
+
+@doc raw"""
+    root_system_type(R::RootSystem)
+
+Return the Dynkin type of the root system `R`.
+"""
+root_system_type(R::RootSystem) = R.root_system_type
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function show(io::IO, R::RootSystem)
+  print(io, "Root system of type ")
+  show(io, string(R.root_system_type[1]) * string(R.root_system_type[2]))
+end
+
+###############################################################################
+#
+#   Comparison
+#
+###############################################################################
+
+function Base.:(==)(R1::RootSystem, R2::RootSystem)
+  return R1.roots == R2.roots && R1.root_system_type != R2.root_system_type
+end
+
+function Base.hash(R::RootSystem, h::UInt)
+  b = 0x9d96557cb5f07773 % UInt
+  h = hash(R.root_system_type, h)
+  h = hash(R.roots, h)
+  return xor(h, b)
+end
+###############################################################################
+#
+#   Constructor
+#
+###############################################################################
+@doc raw"""
+  root_system(S::String) -> root_system
+Return the root system of type `S` where `S` is a string consisting out of
+a letter `A`, `B`, `C`, `D`, `E`, `F`, `G` followed by an integer.
+For the exceptional root system, the integers are fixed, e.g. `G2`, `F4`, `E6`, ...
+"""
```

Comment:
```suggestion
###############################################################################

@doc raw"""
    root_system(S::Symbol, n::Int) -> RootSystem
    
Return the root system of type `Sn` where `S` is a symbol consisting out of
a single letter `A`, `B`, `C`, `D`, `E`, `F`, `G`.
The allowed values for `n` depend on the choice of `S`.
"""
```
Fixes the docstring to be consistent with the function